lib/moralis.ts

```diff
@@ -40,3 +51,35 @@ export const fetchTransactions = async (
     cursor: transfers.data.cursor,
   };
 };
+
+// fetch very first Glo transaction
+export const fetchFirstGloTransaction = async (
+  address: string,
+  chainHex: string
+): Promise<Erc20Transaction | null> => {
+  let cursor = "";
+  do {
+    const response = await Moralis.EvmApi.token.getTokenTransfers({
+      chain: chainHex,
+      address: process.env.NEXT_PUBLIC_USDGLO as string,
+      fromBlock: getFirstGloBlock(hexToNumber(chainHex)),
+      cursor: cursor,
+      limit: 100,
+    });
+
+    const firstTx = response.result.find((tx: Erc20Transaction) => {
```

Comment:
Use https://docs.moralis.io/web3-data-api/evm/how-to-get-all-erc20-transfers-by-wallet#step-2-get-all-erc20-transfers-by-wallet instead? It should allow you to find first tx by wallet?

Reply:
Implemented as part of the latest commit